src/app/Library/CrudPanel/Traits/Fields.php

```diff
@@ -110,6 +110,16 @@ public function addField($field)
         $this->enableTabsIfFieldUsesThem($field);
         $this->addFieldToOperationSettings($field);
 
+        if (! app('UploadStore')->isUploadHandled($field['name'])) {
+            $crudFieldObject = (new CrudField($field['name']));
+
+            foreach ($field as $attribute => $value) {
+                if ($crudFieldObject->hasMacro($attribute)) {
+                    $crudFieldObject->{$attribute}($value);
+                }
+            }
+        }
```

Comment:
Can we please:
- [ ] remove the UploadStore check
- [ ] create a method `CrudField::getMacros()` method that only returns the macro names, and test against that;
- [ ] move this to `$this->callFieldMacros($field)`